checks/startupCheck.go

```diff
@@ -47,3 +67,83 @@ func runCheck(conf *config.Configuration, reg *api.RegistrationResponse, logSend
 func exampleCheckFunction(*config.Configuration, *api.RegistrationResponse) error {
 	return nil
 }
+
+// TODO:
+// validate that java handler ends with interface method handleRequest
+// check on custom runtimes
+// tests
+func checkHandler(conf *config.Configuration, reg *api.RegistrationResponse) error {
+	functionHandler := reg.Handler
+
+	for k, v := range runtimeChecks {
+
+		path := fmt.Sprintf("/var/lang/bin/%s", k)
+
+		_, err := os.Stat(path)
+
+		if err == nil {
+			h := handlerConfigs{
+				handlerName: functionHandler,
+				conf:        conf,
+			}
+			err = v(h)
+			if err != nil {
+				return fmt.Errorf("Lambda handler is set incorrectly: %s", err)
+			}
+			break
+		}
+	}
+	return nil
+}
+
+// Handler format for dotnet functions: Assembly::Namespace.ClassName::Method
+// from the file names alone all that we can validate is the Assembly dll
+func dotnet(h handlerConfigs) error {
+	s := strings.Split(h.handlerName, "::")
+	_, err := os.Stat("/var/task/" + s[0] + ".dll")
+	return err
+}
+
+// Handler format for java functions: package.Class::method
+// we can validate that the package + class path exist
+// and want to validate that the method value is set to ::handleRequest
+func java(h handlerConfigs) error {
+	functionHandler := strings.ReplaceAll(h.handlerName, ".", "/")
+	s := strings.Split(functionHandler, "::")
+	_, err := os.Stat("/var/task/" + s[0] + ".class")
+	return err
+}
+
+// Hander format for node functions: file.function
+// we can validate that the handler has been set to newrelic_lambda_wrapper.handler
+// and that the file name provided in the env var NEW_RELIC_LAMBDA_HANDLER exists
+func node(h handlerConfigs) error {
+	functionHandler, err := checkNrHandler(h)
+	if err != nil {
+		return err
+	}
+	s := strings.Split(functionHandler, ".")
+	_, err = os.Stat("/var/task/" + strings.Join(s[:len(s)-1], "/") + ".js")
```

Comment:
So i moved the path building logic to its own function. It will replace all `::` and `.` characters with `/` and will leave any existing  `/` characters as is. So we should be good for both types of handler formats now.